### PR TITLE
(maint) Handle moved tickets in ticketmatch.rb

### PR DESCRIPTION
**Pull request by smcclellan**
When a ticket has moved in JIRA, but a commit already references that
ticket identifier, the commit was being flagged as unmatched. This is
fixed in this commit by running an extra JIRA query of the old
identifier and mapping the (real) key that gets returned. The script
output will specify that the ticket identifier is an old alias.

### CODE REVIEW

File: ticketmatch.rb [hunk 1]
```diff
@@ -307,7 +311,18 @@ def inspect
...
+      if actual.key?('key') && jira_tickets[actual['key']]
+        jira_tickets.add_alias(actual['key'], ticket)
+        puts "-- #{ticket.upcase} (#{jira_tickets[ticket].state}) [alias of #{actual['key']}]"
+      else
+        puts "** #{ticket.upcase}"
+      end
+    rescue
+      # Failed to look up the ticket.
+      puts "** #{ticket.upcase}"
+    end
```
Comment (speedofdark) [inline ticketmatch.rb:325]: Do you have a copy of the output from our baseline problem that prompted this enhancement?

Comment (scottsolowild) [reply ticketmatch.rb:325]: Here are the relevant sections:
```
<><><><><><><><><><>
Note: fetch origin for puppet-agent...
Checking: puppet-agent
from_rev: 6.0.2, to_rev: 6.0.3_release, fix_ver: 6.0.3
...
** PUP-8351
    72b1f6a6  Update bash wrappers to keep LD_LIBRARY_PATH with RHSCL
...
...
----- Git commits in Jira -----
COMMIT TOKENS NOT FOUND IN JIRA (OR NOT WITH FIX VERSION OF puppet-agent 6.0.3)
PUP-8351
```
This commit now evaluates as:
```
-- PUP-8351 (Closed) [alias of PA-2278]
    72b1f6a6  Update bash wrappers to keep LD_LIBRARY_PATH with RHSCL
```
And will not show up in the "commit tokens not found in JIRA" section.

Comment (speedofdark) [reply ticketmatch.rb:325]: Did the last half not paste?

Comment (scottsolowild) [reply ticketmatch.rb:325]: Updated.

Comment (speedofdark) [reply ticketmatch.rb:325]: Thats excellent!

**Review by speedofdark — APPROVED** [review]

**Comment by mhashizume** [comment]
This PR is almost four years old and neither people involved in this work at Puppet anymore.

Closing this PR in preparation to renaming the default branch.

**Pull request closed without merging**